### PR TITLE
SAM-3271 - Send notifications to all instructors if no instructor is …

### PR DESCRIPTION
**Pull request by danielmerino**
…available in the group the assignment was submitted for.

### CODE REVIEW

File: samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java [hunk 1]
```diff
@@ -224,6 +224,11 @@ private     void                notifyInstructor                    (String site
             
             if (replacementValues.get("releaseToGroups") != null){
             	siteUsersHasRole = extractInstructorsFromGroups(site,replacementValues.get("releaseToGroups") );
+            	if (siteUsersHasRole.isEmpty()){
+            		//If nobody in the group has a maintain role (i.e. a section that can't have instructors or a group withour instructors) use the old site logic.
+            		AuthzGroup azGroup = authzGroupService.getAuthzGroup("/site/" + siteID);
+                	siteUsersHasRole = site.getUsersHasRole(azGroup.getMaintainRole());
+            	}
             }else{
```
Comment (ottenhoff) [inline samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java]: Can you reduce code duplication here and just change this else statement?

**Comment by danielmerino** [comment]
@ottenhoff I have removed duplicate code, though it is less readable imho.

File: samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java [hunk 2]
```diff
@@ -222,8 +222,7 @@ private     void                notifyInstructor                    (String site
             Site            site                    = siteService.getSite(siteID);
             Set<String>     siteUsersHasRole;
             
-            if (replacementValues.get("releaseToGroups") != null){
-            	siteUsersHasRole = extractInstructorsFromGroups(site,replacementValues.get("releaseToGroups") );
+            if ((replacementValues.get("releaseToGroups") != null) && !((siteUsersHasRole=extractInstructorsFromGroups(site,replacementValues.get("releaseToGroups") )).isEmpty())){
```
Comment (ottenhoff) [inline samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java]: you do two separate if statements to make this more readable

Comment (danielmerino) [reply samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java]: Sorry, I don't get it how can I avoid the code duplication with two if statements. The duplicated code is for the two else blocks, but the if blocks are nested. Am I missing something?

Comment (danielmerino) [reply samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java]: it something like this: 
if (A) then
{
	if (B) then doX
	else doY
}
else doY

I don't get how can I avoid duplicating doY.

Comment (ottenhoff) [reply samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java]: https://github.com/sakaiproject/sakai/compare/master...ottenhoff:SAM-3271

**Pull request closed without merging**